src/services/blocks/BlocksService.ts

```diff
@@ -450,84 +474,122 @@ export class BlocksService extends AbstractService {
 		parentHash: Hash,
 		block: Block
 	): Promise<ICalcFee> {
+		const coefficients = api.consts.transactionPayment.weightToFee.map((c) => {
+			return {
+				// Anything that could overflow Number.MAX_SAFE_INTEGER needs to be serialized
+				// to BigInt or string.
+				coeffInteger: c.coeffInteger.toString(10),
+				coeffFrac: c.coeffFrac.toNumber(),
+				degree: c.degree.toNumber(),
+				negative: c.negative,
+			};
+		});
+
+		const parentParentHash: Hash = await this.getParentParentHash(
+			api,
+			parentHash,
+			block
+		);
+
+		const [version, multiplier] = await Promise.all([
+			api.rpc.state.getRuntimeVersion(parentParentHash),
+			api.query.transactionPayment?.nextFeeMultiplier?.at(parentHash),
+		]);
+
+		const specName = version.specName.toString();
+		const specVersion = version.specVersion.toNumber();
+
 		const perByte = api.consts.transactionPayment?.transactionByteFee;
 		const extrinsicBaseWeightExists =
 			api.consts.system.extrinsicBaseWeight ||
 			api.consts.system.blockWeights.perClass.normal.baseExtrinsic;
 
-		let calcFee, specName, specVersion, decorated, runtimeDoesNotMatch;
 		if (
-			perByte === undefined ||
-			extrinsicBaseWeightExists === undefined ||
+			!perByte ||
+			!extrinsicBaseWeightExists ||
+			(this.minCalcFeeRuntime && specVersion < this.minCalcFeeRuntime) ||
 			typeof api.query.transactionPayment?.nextFeeMultiplier?.at !== 'function'
```

Comment:
We try and query this on line 491, so we can should just be able to check multiplier